src/modules/storage/LedgerStorage.ts

```diff
@@ -589,8 +590,49 @@ export class LedgerStorage extends Storages
             });
         }
 
-        function save_output (storage: LedgerStorage, height: Height, tx_idx: number,
-            out_idx: number, hash: Hash, utxo_key: Hash,
+        function is_melting (storage: LedgerStorage, tx: Transaction): Promise<boolean>
+        {
+            return new Promise<boolean>((resolve, reject) =>
+            {
+                if ((tx.type == TxType.Payment) && (tx.inputs.length > 0))
+                {
+                    let utxo = tx.inputs
+                        .map(m => `x'${m.utxo.toBinary(Endian.Little).toString("hex")}'`);
+
+                    let sql =
+                        `SELECT
+                            count(*) as count
+                        FROM
+                            blocks B
+                            INNER JOIN transactions T ON (B.height = T.block_height)
+                            INNER JOIN tx_outputs O ON (T.tx_hash = O.tx_hash)
+                        WHERE
+                            O.type = 1
+                            and O.utxo_key in (${utxo.join(',')})
```

Comment:
```suggestion
                        `SELECT
                            count(*) as count
                        FROM
                            blocks B
                            INNER JOIN transactions T ON (B.height = T.block_height)
                            INNER JOIN tx_outputs O ON (T.tx_hash = O.tx_hash)
                        WHERE
                            O.type = 1
                            and O.used = 0
                            and O.utxo_key in (${utxo.join(',')})
```
Do you need it?

Reply:
I created the function `is_melting`

Reply:
This is necessary for the issue to be resolved.

The type of table tx_outputs can have a different value than the type of table transactions.
The type of transaction in the transactions type.
This function was added from Agora to prevent the refund output of the transaction from being frozen.
Agora has a type in UTXO, and I added it to the table.

Reply:
`and O.used = 0` This condition is required.
However, it can be changed according to the comments being discussed below.

Reply:
It has been modified by adding a table that only manages UTXO.